### ISSUE TITLE
False positive "module has no member" under certain conditions

### ISSUE THREAD

**Issue by lbenezriravin** [PyCQA/pylint#2991]
### Steps to reproduce
```
mkdir foo
touch foo/__init__.py
touch bar.py
```
Run pylint on the following code:
```
import foo

baz_str = ''.join('baz')
foo_baz = getattr(foo, baz_str)
foo_baz.join('123')

baz_str = 'baz'
foo_baz = getattr(foo, baz_str)
foo_baz.join('123')

import bar

baz_str = ''.join('baz')
bar_baz = getattr(bar, baz_str)
bar_baz.join('123')
```

### Current behavior
```
$ pipenv run pylint --disable=W --disable=C test.py 
************* Module test
test.py:5:0: E1101: Module 'foo' has no 'join' member (no-member)
```

pylint is incorrectly identifying `foo_baz` as equivalent to the module `foo`. It does not raise this false positive when the string being passed to `getattr` is not generated with the `join()` method. I'm not certain if other ways of generating the `baz` string will cause this error. Pylint does not raise this false positive when the module in question is a file, rather than a directory.

### Expected behavior

pylint should not raise an error here

### pylint --version output

```
pylint 2.3.1
astroid 2.2.5
Python 3.7.3 (default, May 11 2019, 00:45:16) 
[GCC 8.3.1 20190223 (Red Hat 8.3.1-2)]
```

Thank you for all your work on this great tool!

**Comment by PCManticore** [comment]
Thanks for the report!
